tests/ci/cdk/run-cdk.sh

```diff
@@ -0,0 +1,233 @@
+#!/bin/bash
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set -exuo pipefail
+
+# -e: Exit on any failure
+# -x: Print the command before running
+# -u: Any variable that is not set will cause an error if used
+# -o pipefail: Makes sure to exit a pipeline with a non-zero error code if any command in the pipeline exists with a
+#              non-zero error code.
+
+function delete_container_repositories() {
+  ecr_repos=$(aws ecr describe-repositories)
+  if [[ "${ecr_repos}" == *"${ECR_LINUX_X86_REPO_NAME}"* ]]; then
+    aws ecr delete-repository --repository-name "${ECR_LINUX_X86_REPO_NAME}" --force
+  fi
+}
+
+function destroy_ci() {
+  if [[ "${CDK_DEPLOY_ACCOUNT}" == "838297025124" ]]; then
+    echo "destroy_ci should not be executed on team account."
+    exit 1
+  fi
+  cdk destroy accp-* --force
+  # CDK stack destroy does not delete ecr automatically.
+  delete_container_repositories
+}
+
+function destroy_docker_img_build_stack() {
+  if [[ "${IMG_BUILD_STATUS}" == "Failed" ]]; then
+    echo "Docker images build failed. AWS resources of building Docker images is kept for debug."
+    exit 1
+  fi
+  # Destroy all temporary resources created for all docker image build.
+  cdk destroy accp-docker-image-build-* --force
+}
+
+function create_linux_docker_img_build_stack() {
+  # Clean up build stacks if exists.
+  destroy_docker_img_build_stack
+  # Deploy accp ci stacks.
+  # When repeatedly deploy, error 'EIP failed Reason: Maximum number of addresses has been reached' can happen.
+  # https://forums.aws.amazon.com/thread.jspa?messageID=952368
+  # Workaround: go to AWS EIP console, release unused IP.
+  cdk deploy accp-docker-image-build-linux --require-approval never
```

Comment:
I'm looking into it and apparently the maximum number of addresses is 5. My hypothesis is that the EIP resources weren't being released fast enough, while the cloudformation resources were still being propagated or deleted. I haven't seen this issue come up before though, but I guess this comment was written just in case someone else ran into it.

Reply:
thanks for digging into it. between the workaround cited here and our ability to request soft limit increases, i don't think this should present any meaningful issues moving forward.